pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -193,9 +193,9 @@ type PermissionClaim struct {
 
 	// all claims all resources for the given group/resource.
 	// This is mutually exclusive with resourceSelector.
-	// +required
+	// +optional
 	// +kubebuilder:default=false
-	All bool `json:"all"`
+	All bool `json:"all,omitempty"`
```

Comment:
was originally expecting this as part fo the ResourceSelector.

Related question: what does an empty, but set `resourceSelector` mean?

Reply:
see my question here: https://github.com/kcp-dev/kcp/pull/2222#discussion_r1009281781

Reply:
If we remove the required tag, this leaves space for new field but does not check invariants. The only thing i can think of is adding a CEL validation here?

Reply:
added CEL validation, does this make things look better?

Reply:
> was originally expecting this as part fo the ResourceSelector.

clarified out of band: this is fine, as resource selector is a slice.

Reply:
is there a check that either `All` is true or `len(selectors)>0`?

If I am not missing anything, an empty resource selector and `All==false` is valid now.

Reply:
The rule `self.all != has(self.resourceSelector)` covers that, unit test: https://github.com/kcp-dev/kcp/blob/831cc1a5395cde8de761c8c0745977fef6f7adbf/pkg/apis/apis/v1alpha1/types_apiexport_test.go#L60-L68

Reply:
fixed